safety/safety.py

```diff
@@ -87,17 +88,31 @@ def fetch_database_url(mirror, db_name, key, cached, proxy):
         if cached_data:
             return cached_data
     url = mirror + db_name
-    r = requests.get(url=url, timeout=REQUEST_TIMEOUT, headers=headers, proxies=proxy)
+
+    try:
+        r = requests.get(url=url, timeout=REQUEST_TIMEOUT, headers=headers, proxies=proxy)
+    except requests.exceptions.ConnectionError:
+        raise NetworkConnectionError()
+    except requests.exceptions.Timeout:
+        raise RequestTimeoutError()
+    except requests.exceptions.RequestException:
+        raise DatabaseFetchError()
+
     if r.status_code == 200:
         data = r.json()
         if cached:
             write_to_cache(db_name, data)
         return data
-    elif r.status_code == 403:
+
+    if r.status_code == 403:
         raise InvalidKeyError()
-    elif r.status_code == 429:
+
+    if r.status_code == 429:
```

Comment:
All these condition ranges doesn't look good. There are several codes that would not raise an error. Can you please review the ranges here including those beyond 599 or less than 200?